app/src/main/java/fr/free/nrw/commons/Media.kt

```diff
@@ -77,7 +78,13 @@ class Media constructor(
     var coordinates: LatLng? = null,
     var captions: Map<String, String> = emptyMap(),
     var descriptions: Map<String, String> = emptyMap(),
-    var depictionIds: List<String> = emptyList()
+    var depictionIds: List<String> = emptyList(),
+    /**
+     * This field was added to find non-hidden categories
+     * Stores the mapping of category title to hidden attribute
+     * Example: Mountains => false, CC-BY-SA-2.0 => true
+     */
+    var detailedCategories: Map<String, String> = emptyMap()
```

Comment:
How about making it `Map<String, Boolean>`?
And also (for readibility) rename to `categoriesHiddenStatus`
Finally, please add quotes around String examples to make it clear what is a String example and what is not: `"Mountains" => false, "CC-BY-SA-2.0" => true`

Thanks a lot! :-)

Reply:
@nicolas-raoul I tried adding the Map<String, Boolean> and also added a typeconverter for it but still I was getting 
"error: Cannot figure out how to save this field into database. You can consider adding a type converter for it." build error. That's why I kept it string because there is already a type converter for map<string, string>. Not sure why getting that error

Reply:
Would you mind creating a type converter? I believe it can be done in a few lines. Thanks a lot!

Reply:
@nicolas-raoul as I mentioned above, i already created type converter method in converter.java class still getting this error.

Reply:
> Would you mind creating a type converter? I believe it can be done in a few lines. Thanks a lot!

In the converters.java file I added this new method
```
@TypeConverter
    public static String mapObjectToString2(Map<String,Boolean> objectList) {
        return writeObjectToString(objectList);
    }
```
still getting same error